src/DynamoPackagesUI/Utilities/PackageManagerCommands.cs

```diff
@@ -0,0 +1,479 @@
+                }
+                catch
+                {
+                    MessageBox.Show(String.Format(Resources.MessageFailToUninstallPackage,
+                        ResourceProvider.ProductName,
+                        DownloadRequest.asset_name.ToString()),
+                        Resources.UninstallFailureMessageBoxTitle,
+                        MessageBoxButton.OK, MessageBoxImage.Error);
+                }
+            }
+
+            var settings = Model.PreferenceSettings;
+            PackageDownloadHandle packageDownloadHandle = new PackageDownloadHandle();
+            packageDownloadHandle.Name = DownloadRequest.asset_name;
+            packageDownloadHandle.Done(downloadPath);
+
+            //string installedPkgPath = string.Empty;
+            Package dynPkg = null;
+            if (packageDownloadHandle.Extract(Model, this.PackageInstallPath, out dynPkg))
+            {
+                var p = Package.FromDirectory(dynPkg.RootDirectory, Model.Logger);
+                p.ID = DownloadRequest.asset_id;
+                Application.Current.Dispatcher.Invoke((Action)(() =>
+                {
+                    Loader.Load(p);
+                }));
+                packageDownloadHandle.DownloadState = PackageDownloadHandle.State.Installed;
+                this.PackageInstallPath = string.Empty;
+            }
+        }
+
+        /// <summary>
+        /// Download and Install Dynamo Package
+        /// </summary>
+        /// <param name="pkg"></param>
+        public void DownloadAndInstall(string pkg)
+        {
+            string[] temp = pkg.Split(',');
+            PackageManagerRequest req = new PackageManagerRequest(string.Format("assets/{0}", temp[0]), Method.GET);
+            ResponseWithContentBody<dynamic> response = Client.ExecuteAndDeserializeDynamoRequest(req);
+            DownloadRequest = response.content;
+
+            PackageManagerRequest fileReq = new PackageManagerRequest(string.Format("files/download?file_ids={0}&asset_id={1}", temp[1], temp[0]), Method.GET, true);
+            Response res = Client.ExecuteDynamoRequest(fileReq);
+            var pathToPackage = Client.GetFileFromResponse(res);
+            InstallPackage(pathToPackage);
+        }
+
+        public string GetCustomPathForInstall()
+        {
+            string downloadPath = string.Empty;
+
+            downloadPath = GetDownloadPath();
+
+            if (String.IsNullOrEmpty(downloadPath))
+                return string.Empty;
+
+            PackageInstallPath = downloadPath;
+
+            return downloadPath;
+        }
+
+        private string GetDownloadPath()
+        {
+            var args = new PackagePathEventArgs();
+
+            ShowFileDialog(args);
+
+            if (args.Cancel)
+                return string.Empty;
+
+            return args.Path;
+        }
+
+        private void ShowFileDialog(PackagePathEventArgs e)
```

Comment:
`PromptFileSelectionDialog` could be a better name, as it's prompting user to select file path.